packages/shylu/shylu_node/tacho/src/impl/Tacho_Lapack_Team.hpp

```diff
@@ -32,17 +32,24 @@ template <typename T> struct LapackTeam {
     template <typename MemberType>
     static KOKKOS_INLINE_FUNCTION void potrf_upper(const MemberType &member, const int m, T *__restrict__ A,
                                                    const int as0, const int as1, int *info) {
+      *info = 0;
       if (m <= 0)
         return;
 
       typedef ArithTraits<T> arith_traits;
+      const typename arith_traits::mag_type zero(0);
       for (int p = 0; p < m; ++p) {
         const int jend = m - p - 1;
 
         T *__restrict__ alpha11 = A + (p)*as0 + (p)*as1, *__restrict__ a12t = A + (p)*as0 + (p + 1) * as1,
                         *__restrict__ A22 = A + (p + 1) * as0 + (p + 1) * as1;
 
-        Kokkos::single(Kokkos::PerTeam(member), [&]() { *alpha11 = sqrt(arith_traits::real(*alpha11)); });
+        Kokkos::single(Kokkos::PerTeam(member), [&]() {
+          if (arith_traits::real(*alpha11) <= zero) {
+            *info = 1+p;
+          }
+          *alpha11 = sqrt(arith_traits::real(*alpha11));
```

Comment:
Thank you, @kliegeois, for the comments!! Yes, once we encounter one bad pivot, the rest of the factorization is invalid. So, I would like to catch the first bad pivot in each block and just exit, but I am already summing the index from all the blocks.. The main objective of this PR is to catch the failure with the minimum performance overhead since the failure happens very very rarely. I welcome any suggestions!!

Reply:
If you are interested in the first bad pivot, shoudn't be something like this

```
if (*info == 0 && arith_traits::real(*alpha11) <= zero) {
```
to avoid the overwrite of `*info` if there are multiple `p` for which `arith_traits::real(*alpha11) <= zero`?

Reply:
I did not bother since I sum up the errors from all the teams at the end, but it is probably better that way. Thank you!! Ideally, I may want to exit if info is not zero, but I was worried about deadlock somewhere.

Reply:
I agree that it would be great to exit once info is not zero but you will need synchronization points that could be costly. I would say that if it is unlikely that the algorithm faces a bad pivot, it is probably better not to have an early exit logic.